### PR TITLE
Define `reflect.Manifest` as an `object` rather than a `val`.

### PR DESCRIPTION
**Pull request by sjrd**
Yet another approach, alternative to #7642 and #7643.

Since there is a `trait Manifest`, the term of the same name should be in the same file. Otherwise, in some weird circumstances, like somehow while building Scala.js in the community build (but only in the community build), we get a compile error saying that `Manifest` is already defined in the package object.

Since we cannot define a top-level `val`, we declare it as an `object` instead. This means that we have to explicitly forward all the public members of `ManifestFactory` in `Manifest`.

This is meant as a small incremental fix that does not impact the compiler (since its codegen refers to `ManifestFactory`). Longer term, we should make the codegen use the `object Manifest` instead, and eventually retire `ManifestFactory`.

### CODE REVIEW

**Comment by smarter** [comment]
>  Longer term, we should make the codegen use the object Manifest instead, 

Longer-term we should [kill Manifest with fire](https://github.com/scala/scala/pull/6745), but that's just my opinion :).

**Review by SethTisue — APPROVED** [review]

**Comment by SethTisue** [comment]
note to next reviewer: the context for this is 
https://github.com/scala/community-builds/pull/782#issuecomment-419666368 — we're trying to get Scala.js green in the Scala 2.13 community build

**Comment by sjrd** [comment]
I don't usually do this, but: could we get a second reviewer here sooner rather than later? It will take a few rounds of PR, review, merge PR, review, on Scala.js' side after this PR is merged before we can try Scala.js in the community build again. And all of that really should happen before RC1 is released.

**Comment by SethTisue** [comment]
you can merge it on Friday if no further feedback materializes by then.

**Comment by SethTisue** [comment]
LOL, how were you not already a member of https://github.com/orgs/scala/teams/committers ? I just added you.

**Comment by SethTisue** [comment]
Adriaan can administer the blood oath and conduct the pinky-swearing ceremony in person.

**Comment by SethTisue** [comment]
oh, ha, you were already a member of https://github.com/orgs/scala/teams/admins . well, you might as well be in both, anyway.

**Comment by sjrd** [comment]
Yeah, celebrating the very first time I hit the Merge button in the scala/scala repo :tada: 
```
$ git log --merges --author="Doeraene"
commit 5d118abb206c4bf0ef9fd71b6ec2c58a232ec404 (HEAD -> 2.13.x, scala/2.13.x)
Merge: caf0db9bc6 82f2587e05
Author: Sébastien Doeraene <...>
Date:   Fri Jan 25 11:55:39 2019 +0100

    Merge pull request #7649 from sjrd/manifest-as-object-with-forwarders
    
    Define `reflect.Manifest` as an `object` rather than a `val`.
```
and no other :p

**Pull request merged**
